core/src/test/java/org/infinispan/test/concurrent/InboundRpcSequencerAction.java

```diff
@@ -83,7 +83,7 @@ public void handle(CacheRpcCommand command, Reply reply, DeliverOrder order) {
                   reply.reply(response);
                }
             }, order);
-         } finally {
+         } catch (Throwable t) {
```

Comment:
So the actual cause of double calling is okay?

Reply:
@wburns oops, I missed this comment because the fix got into #7974 and my sync script deleted the branch.

`InboundRpcSequencerAction` was advancing to the "after" states twice: once in the finally block, and again after PerCacheInboundInvocationHandler.handle() asynchronously processed the command. I'm pretty sure #7853 was also needed, because NodeA receives 2 `StateResponseCommand`s.